### ISSUE TITLE
need onnx.svg----train loss converge on 0.04 (so big)

### ISSUE THREAD

**Issue by guge2** [SHU-FLYMAN/HDR-NET#4]
My Gpu has just 8GB memory, so I resize every photo and then train.
I found that my train loss and valid loss are all converge on about 0.04, I don't know why.

![Image](https://github.com/user-attachments/assets/deff6171-0091-4878-a8be-0dc67a4a1166)

Anybody could share a `onnx.svg`? so that we can compare if our HDRNet is the same. 

This is mine (run train.py and open `.onnx` in `https://netron.app`)

```python

if __name__ == '__main__':
    import os
    import onnx
    save_file = "../logs/model.onnx"
    os.makedirs(os.path.dirname(save_file), exist_ok=True)
    h, w = 416, 432  
    data = torch.randn(2, 3, h, w)

    net = ResNetFPN()
    # print(net)
    net.eval()  

    torch.onnx.export(
        net, data, save_file,
        export_params=True, 
        opset_version=11,  
        do_constant_folding=True,
        input_names=['input'],
        output_names=['output']
    )

    print(f"Model exported to {save_file}")

    # optional shape inference
    onnx_model = onnx.load(save_file)
    onnx_model = onnx.shape_inference.infer_shapes(onnx_model)
    onnx.save(onnx_model, save_file)
    print("onnx can open in https://netron.app")
```

![Image](https://github.com/user-attachments/assets/0db23795-14c1-4662-a1d7-cf4a70c5e985)

**Comment by SHU-FLYMAN** [comment]
The majority of the area in this dataset does not have fringe projection, and too low a resolution would cause the phase to be masked by these areas, hence training cannot be conducted at too low a resolution. Additionally, ideal fringe patterns can be used for pre-training first, followed by fine-tuning. I will reorganize and release the weights shortly.


 My apologies for the delay; I've been extremely busy with work.  



---- Replied Message ----
| From | ***@***.***> |
| Date | 4/10/2025 17:01 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | [SHU-FLYMAN/HDR-NET] need onnx.svg----train loss converge on 0.04 (so big) (Issue #4) |

My Gpu has just 8GB memory, so I resize every photo and then train.
I found that my train loss and valid loss are all converge on about 0.04, I don't know why.

Anybody could share a onnx.svg? so that we can compare if our HDRNet is the same.

This is mine (run train.py and open .onnx in https://netron.app)

if__name__=='__main__':
    importosimportonnxsave_file="../logs/model.onnx"os.makedirs(os.path.dirname(save_file), exist_ok=True)
    h, w=416, 432data=torch.randn(2, 3, h, w)

    net=ResNetFPN()
    # print(net)net.eval()  

    torch.onnx.export(
        net, data, save_file,
        export_params=True, 
        opset_version=11,  
        do_constant_folding=True,
        input_names=['input'],
        output_names=['output']
    )

    print(f"Model exported to {save_file}")

    # optional shape inferenceonnx_model=onnx.load(save_file)
    onnx_model=onnx.shape_inference.infer_shapes(onnx_model)
    onnx.save(onnx_model, save_file)
    print("onnx can open in https://netron.app")

model.onnx.png (view on web)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

guge2 created an issue (SHU-FLYMAN/HDR-NET#4)

My Gpu has just 8GB memory, so I resize every photo and then train.
I found that my train loss and valid loss are all converge on about 0.04, I don't know why.

Anybody could share a onnx.svg? so that we can compare if our HDRNet is the same.

This is mine (run train.py and open .onnx in https://netron.app)

if__name__=='__main__':
    importosimportonnxsave_file="../logs/model.onnx"os.makedirs(os.path.dirname(save_file), exist_ok=True)
    h, w=416, 432data=torch.randn(2, 3, h, w)

    net=ResNetFPN()
    # print(net)net.eval()  

    torch.onnx.export(
        net, data, save_file,
        export_params=True, 
        opset_version=11,  
        do_constant_folding=True,
        input_names=['input'],
        output_names=['output']
    )

    print(f"Model exported to {save_file}")

    # optional shape inferenceonnx_model=onnx.load(save_file)
    onnx_model=onnx.shape_inference.infer_shapes(onnx_model)
    onnx.save(onnx_model, save_file)
    print("onnx can open in https://netron.app")

model.onnx.png (view on web)
